File: test-suite/src/test/java/io/micronaut/kafka/docs/producer/config/TransactionalClient.java
package io.micronaut.kafka.docs.producer.config;

import io.micronaut.configuration.kafka.annotation.KafkaClient;
import io.micronaut.context.annotation.Requires;

@Requires(property = "spec.name", value = "TransactionalClientTest")
// tag::clazz[]
@KafkaClient(id = "my-client", transactionalId = "my-tx-id")
public interface TransactionalClient {
    // define client API
}
// end::clazz[]

File: test-suite-python/src/test/python/micronaut/kafka/docs/producer/config/TransactionalClient.py
from abc import ABC

from micronaut.configuration.kafka.annotation import KafkaClient
from micronaut.context.annotation import Requires


@Requires(property="spec.name", value="TransactionalClientTest")
# tag::clazz[]
@KafkaClient(id="my-client", transactionalId="my-tx-id")
class TransactionalClient(ABC):
    # define client API
    pass
# end::clazz[]
